From 1ba823024edddfb0e6ccb92647137eccf5d4c7d0 Mon Sep 17 00:00:00 2001
Message-Id: <1ba823024edddfb0e6ccb92647137eccf5d4c7d0.1557072286.git.liu.denton@gmail.com>
In-Reply-To: <cover.1557072286.git.liu.denton@gmail.com>
References: <cover.1557071877.git.liu.denton@gmail.com>
	<cover.1557072286.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Sun, 21 Apr 2019 22:57:03 -0700
Subject: [PATCH v3 3/4] revisions.txt: mention <rev>~ form
To: Git Mailing List <git@vger.kernel.org>
Cc: Andreas Heiduk <asheiduk@gmail.com>,
    Duy Nguyen <pclouds@gmail.com>,
    Junio C Hamano <gitster@pobox.com>

In revisions.txt, the '<rev>^' form is mentioned but the '<rev>~' form
is missing. Although both forms are essentially equivalent (they each
get the first parent of the specified revision), we should mention the
latter for completeness. Make this change.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
 Documentation/revisions.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Documentation/revisions.txt b/Documentation/revisions.txt
index dd99770c47..a38ec7fb52 100644
--- a/Documentation/revisions.txt
+++ b/Documentation/revisions.txt
@@ -139,7 +139,9 @@ thing no matter the case.
   '<rev>{caret}0' means the commit itself and is used when '<rev>' is the
   object name of a tag object that refers to a commit object.
 
-'<rev>{tilde}<n>', e.g. 'master{tilde}3'::
+'<rev>{tilde}[<n>]', e.g. 'HEAD{tilde}, master{tilde}3'::
+  A suffix '{tilde}' to a revision parameter means the first parent of
+  that commit object.
   A suffix '{tilde}<n>' to a revision parameter means the commit
   object that is the <n>th generation ancestor of the named
   commit object, following only the first parents.  I.e. '<rev>{tilde}3' is
-- 
2.21.0.1049.geb646f7864

